index.js

```diff
@@ -366,9 +368,23 @@ Archive.prototype.appendFile = function (filename, name, cb) {
       mode: st.mode,
       size: 0,
       link: null
-    }, {filename: filename}, cb)
+    }, {filename: filename}, function (err, entry) {
+      if (err) return cb(err)
+      if (!entry) return cb()
+      if (entry.type === 'file') self.readStats.files++
+      cb(err, entry)
+    })
 
-    if (ws) pump(fs.createReadStream(path.resolve(self.directory, filename)), ws)
+    var statsCounter = through(function (data, encoding, next) {
+      self.readStats.totalRead += data.length
+      this.push(data)
+      next()
+    })
+
+    if (ws) {
+      var readStream = fs.createReadStream(path.resolve(self.directory, filename))
+      pump(readStream, statsCounter, ws)
+    }
```

Comment:
@maxogden nvm my comment about tracking other types. i forgot we had the very file explicit api in there so that makes sense